src/coprocessor/dag/expr/builtin_time.rs

```diff
@@ -438,6 +439,24 @@ impl ScalarFunc {
         Ok(Some(Cow::Owned(res)))
     }
 
+    #[inline]
+    pub fn sub_duration_and_string<'a, 'b: 'a>(
+        &'b self,
+        ctx: &mut EvalContext,
+        row: &'a [Datum],
+    ) -> Result<Option<Cow<'a, MyDuration>>> {
+        let arg0: Cow<'a, MyDuration> = try_opt!(self.children[0].eval_duration(ctx, row));
+        let arg1: Cow<'a, [u8]> = try_opt!(self.children[1].eval_string(ctx, row));
+        let s = ::std::str::from_utf8(&arg1)?;
+        let arg1 = MyDuration::parse(&arg1, Time::parse_fsp(s))?;
+        let overflow = Error::overflow("DURATION", &format!("({} + {})", &arg0, &arg1));
```

Comment:
Please move this into line 455 so the `Error` doesn't need to be evaluated even if the subtraction succeed.

Reply:
i have moved it backwards

Reply:
@summer-slow-grace I mean that `format!()` call shouldn't be executed when `checked_sub` returns Some. `overflow` must not be evaluated before `checked_sub`.

Reply:
done